salt/orchestrate/aws/map_templates/edx.yml

```diff
@@ -9,9 +9,17 @@ edx:
           SubnetId: {{ subnetids[id_num % num_subnets] }}
           SecurityGroupId: {{ securitygroupid }}
       tag:
+        business_unit: {{ business_unit }}
         environment: {{ environment_name }}
+        purpose: {{ purpose }}
       grains:
+        business_unit: {{ business_unit }}
         environment: {{ environment_name }}
+        {% if app_type not in purpose %}
+        {% set purpose = purpose[:-len(purpose.split('-')[-1])] +app_type %}
```

Comment:
It's potentially dangerous to overwrite the variable used in the conditional, particularly if it is being used in a loop. This should probably be set to a different name for reference inside the `if` block.